cpp/src/gandiva/precompiled/string_ops_test.cc

```diff
@@ -123,6 +123,26 @@ TEST(TestStringOps, TestSubstring) {
   EXPECT_EQ(std::string(out_str, out_len), "");
   EXPECT_FALSE(ctx.has_error());
 
+  out_str = substr_utf8_int64_int64(ctx_ptr, "अपाचे एरो", 25, 1, 5, &out_len);
```

Comment:
I want that you add tests with invalid values, e.g. 

- invalid utf8 strings (where the invalid char is the first, in the middle, the last)
- very large, e.g. > INT32_MAX, int64 offset values

This function is very security sensitive.